Added label to the host nodes using cephadm for 5.0

preethisn opened the pull request:
# Description -
Added label to the host nodes using cephadm for 5.0
Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
Cephadm enhancement
# Checklist:

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

Logs path - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1611733404859/cephadm_shell_execution_0.log

rakeshgm commented:
@preethisn your PR has lot of old commits, can you update your branch or rebase it properly.  you have around 58 commits.

preethisn commented:
@rakeshgm Rebased and Updated the branch.

psathyan commented on ceph/ceph_admin.py:
```diff
@@ -224,6 +224,73 @@ def manage_hosts(self, **kwargs):
             # todo: add code for removing host
             raise NotImplementedError
 
+    def label_host(self):
```
Please add docstring

preethisn replied:
Added Doc string

psathyan commented on ceph/ceph_admin.py:
```diff
@@ -224,6 +224,73 @@ def manage_hosts(self, **kwargs):
+        node_mds = self.cluster.get_nodes(role="mds")
+        node_rgw = self.cluster.get_nodes(role="rgw")
+        node_client = self.cluster.get_nodes(role="client")
+        # self.ceph_iscsi(self.cluster.get_nodes(role="iscsi"))
```
Could you remove the code that is commented? We can have comments but it would be good to avoid code that is commented.

psathyan commented on ceph/ceph_admin.py:
```diff
@@ -224,6 +224,73 @@ def manage_hosts(self, **kwargs):
+        node_rgw = self.cluster.get_nodes(role="rgw")
+        node_client = self.cluster.get_nodes(role="client")
+        # self.ceph_iscsi(self.cluster.get_nodes(role="iscsi"))
+        for node in node_mon:
```
IMO, we could have a common/private method that takes the node & label.. for example
```
def _apply_label(self, node_name, label):
  self.shell(remote=self.installer, args= [ 'ceph', 'orch', 'host', 'label' , 'add', node_name, label])
```

psathyan commented on tests/ceph_installer/test_cephadm.py:
```diff
@@ -64,6 +64,9 @@ def run(ceph_cluster, **kw):
     cephadm.manage_hosts()
 
+    # Add lables to the host
+    cephadm.label_host()
```
Can we control the flow? In other words, `add_daemons` does not use labels... it would be good if we can modify `add_daemons` to use labels or have an option to to switch between the host or labels. In this way, we can make better use of the labels option. What do you think?

psathyan reviewed (commented):
Would it be possible to run `black` against the modified code? IMHO, it would be a good to have a consistent visual format.

preethisn commented:
@psathyan, I have addressed the review comments based on the discussion, Latest Test log attached - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1611920080519/cephadm_shell_execution_0.log

psathyan commented:
@preethisn #326 would override this PR. Could we discuss with @sunilkumarn417 how to move this forward?
